Makefile

```diff
@@ -25,12 +25,27 @@ install: _output/bin/$(LOCAL_OS)/bootkube
 
 _output/bin/%: $(GOFILES)
 	mkdir -p $(dir $@)
-	GOOS=$(word 1, $(subst /, ,$*)) go build -ldflags "$(LDFLAGS)" -o $@ github.com/kubernetes-incubator/bootkube/cmd/$(notdir $@)
+	GOOS=$(word 1, $(subst /, ,$*)) go build $(GOFLAGS) -ldflags "$(LDFLAGS)" -o $@ github.com/kubernetes-incubator/bootkube/cmd/$(notdir $@)
 
 _output/release/bootkube.tar.gz: _output/bin/linux/bootkube _output/bin/darwin/bootkube _output/bin/linux/checkpoint
 	mkdir -p $(dir $@)
 	tar czf $@ -C _output bin/linux/bootkube bin/darwin/bootkube bin/linux/checkpoint
 
+run-%: GOFLAGS = -i
+run-%: clean-vm-% _output/bin/linux/bootkube _output/bin/$(LOCAL_OS)/bootkube
+	@until (cd hack/$*-node && vagrant up 2>/dev/null);\
+	do \
+		echo "Vagrant not ready."; \
+		sleep 5; \
+	done;
+	@cd hack/$*-node && ./bootkube-up
+	@echo "Bootkube ready"
+
+clean-vm-single:
+clean-vm-%:
+	@echo "Cleaning VM..."
+	@(cd hack/single-node && vagrant destroy -f && vagrant up) &
```

Comment:
Might also want to add `rm -rf /cluster` -- otherwise you could have compiled bootkube with new asset templates, but `bootkube-up` will not overwrite them if directory already exists.